### ISSUE TITLE
Improve consuming generators

### ISSUE THREAD

**Issue by rsokl** [rsokl/Learning_Python#91]
Add more examples than just using `sum` to consume a generator. Also fix [the solution](https://www.pythonlikeyoumeanit.com/Module2_EssentialsOfPython/Generators_and_Comprehensions.html#Reading-Comprehension-Exercise-Solutions) to actually show the iteration and printing. Currently things are very confusing.

**Comment by alexding123** [comment]
Getting an error here...
```bash
$ conda env create -f environment.yml

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/anaconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1043, in __call__
        return func(*args, **kwargs)
      File "/anaconda3/lib/python3.7/site-packages/conda_env/cli/main.py", line 73, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/anaconda3/lib/python3.7/site-packages/conda_env/cli/main_create.py", line 77, in execute
        directory=os.getcwd())
      File "/anaconda3/lib/python3.7/site-packages/conda_env/specs/__init__.py", line 40, in detect
        if spec.can_handle():
      File "/anaconda3/lib/python3.7/site-packages/conda_env/specs/yaml_file.py", line 18, in can_handle
        self._environment = env.from_file(self.filename)
      File "/anaconda3/lib/python3.7/site-packages/conda_env/env.py", line 144, in from_file
        return from_yaml(yamlstr, filename=filename)
      File "/anaconda3/lib/python3.7/site-packages/conda_env/env.py", line 130, in from_yaml
        data = validate_keys(data, kwargs)
      File "/anaconda3/lib/python3.7/site-packages/conda_env/env.py", line 35, in validate_keys
        for key in data.keys():
    AttributeError: 'NoneType' object has no attribute 'keys'
```

**Comment by rsokl** [comment]
Closed by #93
